.github/workflows/build-test-upload.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Dynatrace LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build, test and upload
+on:
+  pull_request:
+    types: [ 'closed' ]
```

Comment:
why not just main branch?

Reply:
So that the artifact is only published once the PR is closed

Reply:
What if PR is closed without merging to main branch? Shouldn't we trigger upload only on main branch